src/test/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMappingTest.java

```diff
@@ -385,22 +385,109 @@ void getHearingLocations() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        setupAppeal(caseData);
+        setupAdjournedHearingVenue(caseData, "sameVenue");
+
+        caseData.setHearings(Collections.singletonList(Hearing.builder()
+                    .value(uk.gov.hmcts.reform.sscs.ccd.domain.HearingDetails.builder()
+                    .venueId(HEARING_VENUE_ID_OTHER).build())
+                .build()));
+
+        checkHearingLocationResults(HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder),
+                                    HEARING_VENUE_ID_OTHER);
+    }
+
+    @DisplayName("When a case has been adjourned but the next hearing is paper, return the override hearing locations")
+    @Test
+    void getHearingLocationsAdjournmentNewVenuePaperCase() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        setupAdjournedHearingVenue(caseData, "somewhereElse");
+
+        checkHearingLocationResults(HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder),
+                                    HEARING_VENUE_ID_CURRENT, HEARING_VENUE_ID_OTHER);
+    }
+
+    @DisplayName("Checks both the errors we can throw when trying to obtain the venue ID when getting the locations")
+    @Test
+    void getHearingLocationsFailOnGettingVenueId() {
+        SscsCaseData caseData = buildOverrideHearingLocations();
+
+        setupAppeal(caseData);
+
+        String unknownValue = "UnknownValue";
+        setupAdjournedHearingVenue(caseData, "UnknownValue");
+
+        assertThatThrownBy(() -> HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessageContaining("Failed to determine next hearing venue: " + unknownValue);
+
+        caseData.setAdjournCaseNextHearingVenue("sameVenue");
+
+        assertThatThrownBy(() -> HearingsDetailsMapping.getHearingLocations(caseData, referenceDataServiceHolder))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessageContaining("Failed to determine next hearing venue due to no latest hearing");
+    }
+
+    void setupAdjournedHearingVenue(SscsCaseData caseData, String nextHearingVenue) {
+        DynamicListItem item = new DynamicListItem(HEARING_VENUE_ID_CURRENT, "");
+        DynamicList list = new DynamicList(item, Collections.emptyList());
+
+        caseData.setAdjournCaseNextHearingVenue(nextHearingVenue);
+        caseData.setAdjournCaseNextHearingVenueSelected(list);
+    }
+
+    void checkHearingLocationResults(List<HearingLocation> hearingLocations, String... expectedResults) {
+        assertThat(hearingLocations)
+            .hasSize(expectedResults.length)
+            .extracting("locationId","locationType")
+            .containsExactlyInAnyOrder(Arrays.stream(expectedResults)
+                .map(result -> new Tuple(result, COURT)).toArray(Tuple[]::new));
```

Comment:
Good work, but there is a cleaner way of doing this:
```suggestion
            .allSatisfy(hearingLocation ->
                assertThat(hearingLocation.getLocationType).isEqualTo(COURT))
            .extracting(HearingLocation::getLocationId)
            .containsExactlyInAnyOrder(expectedResults);
```